cmd/cluster/sre_operators/listCmd.go

```diff
@@ -0,0 +1,475 @@
+package sre_operators
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/google/go-github/v63/github"
+	"github.com/openshift/osdctl/pkg/printer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/xanzy/go-gitlab"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/cmd/util"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type sreOperatorsListOptions struct {
+	short     bool
+	outdated  bool
+	noHeaders bool
+	operator  string
+
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+type sreOperator struct {
+	Name     string
+	Current  string
+	Expected string
+	Status   string
+	Channel  string
+}
+
+var (
+	RestoreColor = "\033[0m"
+	Red          = "\033[31m"
+	Bold         = "\u001b[1m"
+	Gap          = "         "
+)
+
+const (
+	sreOperatorsListExample = `
+	# List SRE operators
+	$ osdctl cluster sre-operators list
+	
+	# List SRE operators without fetching the latest version for faster output
+	$ osdctl cluster sre-operators list --short
+	
+	# List only SRE operators that are running outdated versions
+	$ osdctl cluster sre-operators list --outdated
+
+	# List a specific SRE operator
+	$ osdctl cluster sre-operators list --operator='OPERATOR_NAME'
+	`
+	sreOperatorsListDescription = `
+  Lists the current version, channel, and status of SRE operators running in the current 
+  cluster context, and by default fetches the latest version from the operators' repositories.
+  
+  A GitLab access token is required to fetch the latest version of the operators,
+  and can be set in the '.config/osdctl' file as 'gitlab_access: <TOKEN>'.
+
+  The command creates a Kubernetes client to access the current cluster context, and GitLab/GitHub
+  clients to fetch the latest versions of each operator from its respective repository.
+	`
+	repositoryBranch = "production"
```

Comment:
This would make the command specific to prod only. We can try to fetch the environment from OCM directly or add a note that this command would only be relevant to Prod and should actually fail in Stg/Int

Reply:
Unfortunately not all of the operators' repositories obey the stg/int/prod branching format, so it wouldn't be possible to correctly fetch versions based on this. The command still works in stg/int, with only a select few operators displaying different versions, although this is made clear by the 'channel' column of the command (i.e. on stg an operator using staging channel will be shown as such)